### PR TITLE
Swift 5 fixes

### PR DESCRIPTION
**Pull request by dylanmaryk**
This fixes errors and warnings when building Cuckoo with Swift 5.

### CODE REVIEW

File: Source/MockManager.swift [hunk 1]
```diff
@@ -179,11 +179,11 @@ extension MockManager {
 
 extension MockManager {
     public func getter<T>(_ property: String, superclassCall: @autoclosure () -> T, defaultCall: @autoclosure () -> T) -> T {
-        return call(getterName(property), parameters: Void(), escapingParameters: Void(), superclassCall: superclassCall, defaultCall: defaultCall())
+        return call(getterName(property), parameters: Void(), escapingParameters: Void(), superclassCall: superclassCall(), defaultCall: defaultCall())
```
Comment (ApurvaGaikwad) [inline Source/MockManager.swift]: @dylanmaryk : Getting error after updating pod

Screenshot 2019-04-05 at 6 42 11 PM

Add () to forward @autoclosure parameter

Comment (dylanmaryk) [reply Source/MockManager.swift]: The '()' is what I put in my PR, but it's not there in your screenshot for some reason? The error you're getting is exactly what I fixed.

Comment (ApurvaGaikwad) [reply Source/MockManager.swift]: Still it is showing same error after updating

Comment (dylanmaryk) [reply Source/MockManager.swift]: But after you apply my changes, are the errors fixed?

Comment (ApurvaGaikwad) [reply Source/MockManager.swift]: After pushing changes on gitlab it is showing me error like this: 

Screenshot 2019-04-08 at 1 02 03 PM

Comment (ApurvaGaikwad) [reply Source/MockManager.swift]: @dylanmaryk Can you please merge this change in master & release new updated pod.

Comment (dylanmaryk) [reply Source/MockManager.swift]: I don't have the permissions required to merge the change. It hasn't even been approved yet.

Comment (MatyasKriz) [reply Source/MockManager.swift]: @ApurvaGaikwad can't you just use the `swift-5-fixes` branch for now?

**Comment by TadeasKriz** [comment]
@dylanmaryk is this change backward compatible with Swift 4?

**Comment by TadeasKriz** [comment]
Also, could you check if the generated code is compilable if it contains `@autoclosure`? I don't see any changes in the Generator so I just want to be sure. Thanks!

**Comment by dylanmaryk** [comment]
@TadeasKriz yes it's compiling with Swift 4.2. I've had no problems with the generated code. `@autoclosure` was already there too, I didn't add it.

File: Source/MockManager.swift (hunk 1 above)
Comment (ApurvaGaikwad) [reply Source/MockManager.swift]: @MatyasKriz @dylanmaryk : Thanks for the reply. I have noticed the changes but these changes are required to run on the pipeline using the script and for this no manual changes are/should be pushed for Cuckoo.

**Comment by sharaquss** [comment]
@TadeasKriz Is there any progress with this PR? This _autoclosure_ error is making Cuckoo unusable with Swift 5. Thanks!

**Comment by TadeasKriz** [comment]
Merged in 6c55017. Thanks!

**Pull request closed without merging**